### PR TITLE
[WIP] [L1] Matic deployment

### LINKED ISSUE #25: Adapt nucypher-ts to accept Matic (and other L2s) RPC endpoints
@piotr-roslaniec:
- Adding support for EVM-compatible L1s is going to be as easy as adding contract and network configuration for a new network.
- `nucypher-ts` already works on Mumbai (Polygon testnet) in some limited capacity (policy creation). Further work needs to be done in other areas of the network (Ursulas, Porter) to enable complete user flows such as granting and revocation of policies.
- In order to deploy `nucypher-ts` on Mumbai we had to rewrite some of the policy validation logic - The logic of `SubscriptionManager::createPolicy` is somewhat different than `PolicyManager::createPolicy` 
- This issue is blocked by the current incompatibility with `nucypher`
- Related: #31

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
* [ ] Update policy revocation
* [x] Update policy parameter generation
* [ ] Fix missing contract implementation

Changes:
- Replace previous contracts with `StakingEscrow`
- Remove generated types from git repository, generate `typechain` types automatically
- Bump `typechain` version

Closes #25

### CODE REVIEW

File: src/agents/constants.ts [hunk 1]
```diff
@@ -1,37 +1,59 @@
...
-export const CONTRACTS: Record<ChecksumAddress, Contracts> = {
-  mainnet: MAINNET,
-  goerli: GOERLI,
-  rinkeby: RINKEBY,
+const RINKEBY: Partial<ContractAddress> = {
+  STAKING_ESCROW: '0x6A6F917a3FF3d33d26BB4743140F205486cD6B4B',
+};
+
+const MUMBAI: Partial<ContractAddress> = {
+  STAKING_ESCROW: '0x6A6F917a3FF3d33d26BB4743140F205486cD6B4B',
```
Comment (KPrasch) [inline src/agents/constants.ts:26]: Interesting - what are your thoughts regarding the use of StakingEscrow on mumbai?

Comment (piotr-roslaniec) [reply src/agents/constants.ts:26]: Oh yeah, it's not going to be used I think,  `SubscriptionManager` should be sufficient in any case. `StakingEscrow` is not even deployed in Mumbai, I included it by accident.

File: src/agents/constants.ts [hunk 2]
```diff
@@ -1,37 +1,58 @@
...
-export const CONTRACTS: Record<ChecksumAddress, Contracts> = {
-  mainnet: MAINNET,
-  goerli: GOERLI,
-  rinkeby: RINKEBY,
+const RINKEBY: Partial<ContractAddress> = {
+  STAKING_ESCROW: '0x6A6F917a3FF3d33d26BB4743140F205486cD6B4B',
+};
+
+const MUMBAI: Partial<ContractAddress> = {
+  SUBSCRIPTION_MANAGER: '0x5029dA19035f24b04229dBc6e73463b015F5E071',
```
Comment (KPrasch) [inline src/agents/constants.ts:26]: What address is this?  The `SubscriptionManager` is deployed to Mumbai here at 0x4a5C685fEBfdBB4855EF0E0ac34580dbAD531a21.

https://mumbai.polygonscan.com/address/0x4a5C685fEBfdBB4855EF0E0ac34580dbAD531a21

Comment (piotr-roslaniec) [reply src/agents/constants.ts:26]: I deployed it myself before this one went up. I think it's going to be replaced anyway after we ship on-chain revocation.

**Pull request closed without merging**